contributors/design-proposals/statefulset-update.md

```diff
@@ -0,0 +1,826 @@
+    // RollingUpdateStatefulSetStrategyType indicates that update will be 
+    // applied to all Pods in the StatefulSet with respect to the StatefulSet 
+    // ordering constraints. When a scale operation is performed with this 
+    // strategy, new Pods will be created from the updated specification.
+    RollingUpdateStatefulSetStrategyType = "RollingUpdate"
+    // OnDeleteStatefulSetStrategyType triggers the legacy behavior. Version 
+    // tracking and ordered rolling restarts are disabled. Pods are recreated 
+    // from the StatefulSetSpec when they are manually deleted. When a scale 
+    // operation is performed with this strategy, new Pods will be created 
+    // from the current specification.
+    OnDeleteStatefulSetStrategyType = "OnDelete"
+)
+
+// PartitionStatefulSetStrategy contains the parameters used with the 
+// PartitionStatefulSetStrategyType.
+type PartitionStatefulSetStrategy struct {
+    // Ordinal indicates the ordinal at which the StatefulSet should be 
+    // partitioned.
+    Ordinal int32
+}
+
+type StatefulSetSpec struct {
+    // Replicas, Selector, Template, VolumeClaimsTemplate, and ServiceName 
+    // omitted for brevity.
+
+	// UpdateStrategy indicates the StatefulSetUpdateStrategy that will be 
+	// employed to update Pods in the StatefulSet when a revision is made to 
+	// Template or VolumeClaimsTemplate.
+	UpdateStrategy StatefulSetUpdateStrategy `json:"updateStrategy,omitempty`
+
+	// RevisionHistoryLimit is the maximum number of revisions that will 
+	// be maintained in the StatefulSet's revision history. The revision history
+	// consists of all revisions not represented by a currently applied 
+	// StatefulSetSpec version. The default value is 2.
+	RevisionHistoryLimit *int32 `json:revisionHistoryLimit,omitempty`
+}
+```
+
+The following modifications will be made to the StatefulSetStatus API object.
+
+```go
+ type StatefulSetStatus struct {
+    // ObservedGeneration and Replicas fields are omitted for brevity.
+
+ 	// CurrentRevision, if not empty, indicates the version of PodSpecTemplate, 
+ 	// VolumeClaimsTemplate tuple used to generate Pods in the sequence
+ 	// [0,CurrentReplicas).
+ 	CurrentRevision string `json:"currentRevision,omitempty"`
+ 	
+ 	// UpdateRevision, if not empty, indicates the version of PodSpecTemplate, 
+    // VolumeClaimsTemplate tuple used to generate Pods in the sequence
+    // [Replicas-UpdatedReplicas,Replicas)
+ 	UpdateRevision string `json:"updateRevision,omitempty"`
+ 	
+ 	// ReadyReplicas is the current number of Pods, created by the StatefulSet
+    // controller, that have a Status of Running and a Ready Condition.
+ 	ReadyReplicas int32 `json:"readyReplicas,omitempty"`
+ 	
+ 	// CurrentReplicas is the number of Pods created by the StatefulSet 
+    // controller from the PodTemplateSpec, VolumeClaimsTemplate tuple indicated 
+    // by CurrentRevision.
+ 	CurrentReplicas int32 `json:"currentReplicas,omitempty"`
```

Comment:
We have had multiple discussions over the naming and have landed on this for consistency with DaemonSet. I don't see a reason to change the naming yet again.